fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/client/BulkDataClient.java

```diff
@@ -312,11 +309,139 @@ public PollingLocationResponse status(String job) throws Exception {
+        //     B - SC_ACCEPTED (202) GO_TO ACCEPTED
+        Response.Status status = Response.Status.NO_CONTENT;
+        try {
+            // NOT_LOCAL - follow location 
+            String baseUrl =
+                    properties.get(BulkDataConfigUtil.BATCH_URL).replace("jobinstances", "jobexecutions") + "/" + job
+                            + "?action=stop";
+            WebTarget target = getWebTarget(baseUrl);
+
+            // The documentation says this is a PUT, however in practice it's actually a get.
+            // @see https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_batch_rest_api.html#rwlp_batch_rest_api__http_return_codes
+            Response r = target.request().get();
+            String responseStr = r.readEntity(String.class);
+
+            if (HttpStatus.SC_MOVED_TEMPORARILY == r.getStatus()) {
+                // It's on a different server, and must be in the cluster.
+                // If not, we're going to connect to location, and try a put.
+                String location = r.getHeaderString("location");
+                if (location != null) {
+                    // No matter what, tell people we accepted the call.
+                    target = getWebTarget(location + "?action=stop");
+                    // no assignment intentionally.
+                    target.request().get();
+                }
+                // Here, we could easily respond with an Exception/OperationOutcome, however An unexpected error has 
+                // occurred while stopping/deleting the job on a batch cluster member
+                status = Response.Status.ACCEPTED;
+            } else if (HttpStatus.SC_OK == r.getStatus()) {
+                // Check if the Status is GOOD
+                JobExecutionResponse response = JobExecutionResponse.Parser.parse(responseStr);
+                if (response.getBatchStatus().contains("STOPPING")) {
+                    // Signal that we're in a stopping condition.
+                    status = Response.Status.ACCEPTED;
+                }
+                // Don't Delete, let the client flow through again and DELETE.
+            } else if (HttpStatus.SC_CONFLICT == r.getStatus()) {
+                status = Response.Status.ACCEPTED;
+            } else {
+                // Error Condition
+                throw BulkDataExportUtil.buildOperationException(
+                        "An unexpected error has ocurred while stopping/deleting the job", IssueType.TRANSIENT);
+            }
+
+        } catch (FHIROperationException ex) {
+            throw ex;
+        } catch (Exception ex) {
+            throw BulkDataExportUtil.buildOperationException("An unexpected error has ocurred while deleting the job",
+                    IssueType.TRANSIENT);
+        }
+        return status;
+    }
+
+    /*
+     * deletes the job
+     */
+    private Response.Status runDeleteOfTenantForJob(String job) throws Exception {
```

Comment:
maybe better to change the function name...

Reply:
changed